transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringDatagramChannel.java

```diff
@@ -451,43 +472,103 @@ protected void readComplete0(int res) {
         }
 
         @Override
-        protected void scheduleRead0() {
+        protected int scheduleRead0() {
             final IOUringRecvByteAllocatorHandle allocHandle = recvBufAllocHandle();
             ByteBuf byteBuf = allocHandle.allocate(alloc());
-            IOUringSubmissionQueue submissionQueue = submissionQueue();
-
             assert readBuffer == null;
             readBuffer = byteBuf;
 
-            recvMsg = !isConnected();
-            long bufferAddress = byteBuf.memoryAddress();
-            allocHandle.attemptedBytesRead(byteBuf.writableBytes());
+            int writable = byteBuf.writableBytes();
+            allocHandle.attemptedBytesRead(writable);
+            int datagramSize = config().getMaxDatagramPayloadSize();
+
+            int numDatagram = datagramSize == 0 ? 1 : Math.max(1, byteBuf.writableBytes() / datagramSize);
+
+            if (isConnected() && numDatagram <= 1) {
```

Comment:
if the connection is true, a read event will be submitted and [here](https://github.com/netty/netty/blob/write_read_batching/transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringDatagramChannel.java#L507) a recvmsg event, why is that?

Reply:
sorry I can't follow... can you elaborate ?

Reply:
[scheduleRecvmsg0](https://github.com/netty/netty/blob/write_read_batching/transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringDatagramChannel.java#L508) will be executed when isConnected is false, why is a read event submitted when isConnected is true?

Reply:
because generally speaking `read` is cheaper then `recvmsg`.